python/cugraph/cugraph/gnn/pyg_extensions/data/cugraph_store.py

```diff
@@ -0,0 +1,737 @@
+            if isinstance(elem, (tuple, list)):
+                return cls(*elem)
+            if isinstance(elem, dict):
+                return cls(**elem)
+        return cls(*args, **kwargs)
+
+
+class EXPERIMENTAL__CuGraphStore:
+    """
+    Duck-typed version of PyG's GraphStore.
+    """
+    def __init__(self, G, backend='torch'):
+        """
+            G : PropertyGraph or MGPropertyGraph
+                The cuGraph property graph where the
+                data is being stored.
+            backend : The backend that manages tensors (default = 'torch')
+                Should usually be 'torch' ('torch', 'cupy' supported).
+        """
+
+        if backend == 'torch':
+            from torch.utils.dlpack import from_dlpack
+            from torch import int64 as vertex_type
+            from torch import float32 as property_type
+        elif backend == 'cupy':
+            from cupy import from_dlpack
+            from cupy import int64 as vertex_type
+            from cupy import float32 as property_type
+        else:
+            raise ValueError(f'Invalid backend {backend}.')
+        self.backend = backend
+        self.from_dlpack = from_dlpack
+        self.vertex_type = vertex_type
+        self.property_type = property_type
+
+        self.__graph = G
+        self.__subgraphs = {}
+
+        self.__edge_type_lookup_table = G.get_edge_data(
+            columns=[
+                G.src_col_name,
+                G.dst_col_name,
+                G.type_col_name
+            ]
+        )
+
+        self.__edge_types_to_attrs = {}
+        for edge_type in self.__graph.edge_types:
+            edges = self.__graph.get_edge_data(types=[edge_type])
+            dsts = edges[self.__graph.dst_col_name].unique()
+            srcs = edges[self.__graph.src_col_name].unique()
+
+            if self.is_mg:
+                dsts = dsts.compute()
+                srcs = srcs.compute()
+
+            dst_types = self.__graph.get_vertex_data(
+                vertex_ids=dsts,
+                columns=['_TYPE_']
+            )._TYPE_.unique()
+
+            src_types = self.__graph.get_vertex_data(
+                vertex_ids=srcs,
+                columns=['_TYPE_']
+            )._TYPE_.unique()
+
+            if self.is_mg:
+                dst_types = dst_types.compute()
+                src_types = src_types.compute()
+
+            err_string = (
+                f'Edge type {edge_type} associated'
+                'with multiple src/dst type pairs'
+            )
+            if len(dst_types) > 1 or len(src_types) > 1:
+                raise TypeError(err_string)
+
+            pyg_edge_type = (src_types[0], edge_type, dst_types[0])
+
+            self.__edge_types_to_attrs[edge_type] = CuGraphEdgeAttr(
+                edge_type=pyg_edge_type,
+                layout=EdgeLayout.COO,
+                is_sorted=False,
+                size=len(edges)
+            )
+
+            self.__dict__['_edge_attr_cls'] = CuGraphEdgeAttr
```

Comment:
For some reason, this is how the PyG code did it but I think since we're not using some of the features their code uses, we don't need to access attributes through the __dict__.  I accepted your commit.